external_accounts/views.py

```diff
@@ -157,4 +159,50 @@ def citesphere_disconnect(request, repository_id):
             'message': 'Error disconnecting Citesphere account. Please try again or contact support if the issue persists.'
         })
 
+    return redirect(reverse('dashboard'))
+
+@login_required
+def conceptpower_login(request):
+    if request.method == 'POST':
+        username = request.POST.get('username')
+        password = request.POST.get('password')
+        next_url = request.POST.get('next', reverse('dashboard'))
+
+        if not username or not password:
+            messages.error(request, "Both username and password are required.")
+            return render(request, 'login/login.html')
+
+        account, created = ConceptpowerAccount.objects.get_or_create(user=request.user)
+        account.username = username
+        account.password = password
+        account.save()
+        messages.success(request, "Successfully connected to ConceptPower!")
+        return redirect(next_url)
+    return render(request, 'login/login.html')
+    
+@login_required
+def conceptpower_update_password(request):
+    if request.method == 'POST':
+        new_password = request.POST.get("new_password")
+
+        try:
+            account = ConceptpowerAccount.objects.get(user=request.user)
+            account.password = new_password
+            account.save()
+            messages.success(request, "Password updated successfully!")
+        except ConceptpowerAccount.DoesNotExist:
+            messages.error(request, "No ConceptPower account found.")
+        except Exception as e:
+            messages.error(request, "Something went wrong! Please try again.")
+            print(f"Error updating ConceptPower password: {e}")
+
+    return redirect(reverse('dashboard'))
+
+@login_required
+def conceptpower_disconnect(request):
+    try:
+        ConceptpowerAccount.objects.filter(user=request.user).delete()
+    except Exception as e:
+        print(f"Error disconnecting Conceptpower account: {e}")
+        
```

Comment:
the user should see some message saying that either the account was disconnected successfully or that it didn't work.